drivers/storage/portworx/portworx.go

```diff
@@ -88,7 +89,67 @@ func (p *portworx) Init(
 	return nil
 }
 
-func (p *portworx) Validate() error {
+func (p *portworx) Validate(cluster *corev1.StorageCluster) error {
+	podSpec, err := p.GetStoragePodSpec(cluster, "")
+	if err != nil {
+		return err
+	}
+
+	preFlighter := NewPreFlighter(cluster, p.k8sClient, podSpec)
+
+	// Start the pre-flight container. The pre-flight checks at this time are specific to enabling DMthin
+	err = preFlighter.RunPreFlight()
+	if err != nil {
+		if errors.IsAlreadyExists(err) {
+			logrus.Debugf("pre-flight: pre-flight container already running...")
+		}
+		return err
+	}
+
+	cnt := 0
+	//  Wait for all the pre-flight pods to finish
+	for {
+		time.Sleep(3 * time.Second) // Pause before status check
+		completed, inProgress, total, err := preFlighter.GetPreFlightStatus()
+		if err != nil {
+			logrus.Errorf("pre-flight: error getting pre-flight status: %v", err)
+			return err
+		}
+		logrus.Infof("pre-flight: Completed [%v] In Progress [%v] Total [%v]", completed, inProgress, total)
+
+		if total != 0 && completed == total {
+			logrus.Infof("pre-flight: completed...")
+			break
+		}
+
+		// Add five minute timeout.  If we do reconcile loop check we will need a different way.
+		cnt++
+		if cnt == 100 { // 3s * 100 = 300s (5 mins)
```

Comment:
This may not be enough because the first time we would be pulling the px-enterprise image as well. Should we set it to 10mins?

Reply:
incremented the time for now and logged https://portworx.atlassian.net/browse/PWX-30419 to fix this.